extensions/amp-accordion/1.0/base-element.js

```diff
@@ -0,0 +1,267 @@
+/**
+ * Copyright 2021 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as Preact from '../../../src/preact';
+import {
+  Accordion,
+  AccordionContent,
+  AccordionHeader,
+  AccordionSection,
+} from './component';
+import {PreactBaseElement} from '../../../src/preact/base-element';
+import {
+  childElementsByTag,
+  dispatchCustomEvent,
+  toggleAttribute,
+} from '../../../src/dom';
+import {dict, memo} from '../../../src/utils/object';
+import {forwardRef} from '../../../src/preact/compat';
+import {getExpandStateTrigger} from './amp-accordion';
+import {pureDevAssert} from '../../../src/pure-assert';
+import {toArray, toWin} from '../../../src/types';
+import {
+  useImperativeHandle,
+  useLayoutEffect,
+  useRef,
+} from '../../../src/preact';
+import {useSlotContext} from '../../../src/preact/slot'; // DO NOT SUBMIT
+
+const SECTION_SHIM_PROP = '__AMP_S_SHIM';
+const HEADER_SHIM_PROP = '__AMP_H_SHIM';
+const CONTENT_SHIM_PROP = '__AMP_C_SHIM';
+const SECTION_POST_RENDER = '__AMP_PR';
+const EXPAND_STATE_SHIM_PROP = '__AMP_EXPAND_STATE_SHIM';
+
+/** @extends {PreactBaseElement<AccordionDef.AccordionApi>} */
+export class BaseElement extends PreactBaseElement {
+  /** @override */
+  init() {
+    this.registerApiAction('toggle', (api, invocation) =>
```

Comment:
I believe this is to tie into the AMP event system?

Reply:
Checking the generator template it looks like this indeed should live in the AMP code. Updated.